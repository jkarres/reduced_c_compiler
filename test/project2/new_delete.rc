structdef S {
  int i; float f;
};

function : void main() {
  S* ps = NULL;

  cout << ps == NULL << endl;

  new ps;

  cout << ps == NULL << endl;

  ps->i = 51;
  ps->f = 97.34;

  cout << ps->i << endl;
  cout << ps->f << endl;

  delete ps;

  cout << ps == NULL << endl;

}
