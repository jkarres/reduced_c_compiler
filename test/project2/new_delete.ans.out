true
false
51
97.34
true
